### DISCUSSION TITLE
Streetmerchant in Docker

### DISCUSSION THREAD

**Discussion by R3PTIL3** [jef/streetmerchant#1274] [Help] [answered]
Does anyone have instructions to install and run streetmerchant in docker? (Synology)

**Comment by OvertCoffee** [comment, marked as answer]
https://jef.codes/streetmerchant/getting-started/ has examples for docker nightly and notification tests.

**Reply by R3PTIL3** [reply]
Yes but only „HowTo run“ Not install? Noob Tutorial pls. Start Docker.... and? 😁

**Reply by OvertCoffee** [reply]
Tl;dr:

1. Start Docker
2. In a terminal (cmd/powershell/bash, etc.) ```cd``` to the directory that contains your environment file (.env)
3. Type ```docker run --cap-add=SYS_ADMIN -it --rm --env-file ./.env ghcr.io/jef/streetmerchant:nightly``` into your terminal
4. Press enter

I'm going to mangle this but, basically, that's how Docker works. Somewhere out in cloud land (in this case at ghcr.io/jef/streetmerchant) there's a Docker image file that was built for a specific purpose/application. Docker gets a copy of that image (the one tagged "nightly" per the line from the guide) then does some containers-not-VMs magic to make that image (the streetmerchant application) run on your system, inside a container, exactly how it ran on the system used to build the image. 

The docker run command will look for a local copy of the image, if one doesn't exist it'll pull a copy. That's about as close to "installing" as it gets in this context. The ```-it``` flag in that ```docker run``` command will open an interactive tty (command-line) into the running container, which lets you see and interact with streetmerchant like you could if you ran it via node instead of with Docker.

Other args:

```--rm``` tells Docker to delete (remove) the container when you shut it down

```--env-file ./.env``` tells Docker to use the file named ".env" in the current directory for environment variables, which is how you configure options in streetmerchant.

```--cap-add=SYS_ADMIN``` gives the container *basically* root access to your host machine (the one you installed Docker on, that hosts your containers). Seems this is necessary for reasons I don't even pretend to understand, see #1291
